lib/internal/streams/destroy.js

```diff
@@ -50,10 +50,12 @@ function destroy(err, cb) {
   this._destroy(err || null, (err) => {
     const emitClose = (w && w.emitClose) || (r && r.emitClose);
     if (cb) {
+      // Invoke callback before scheduling emitClose so that callback
+      // can schedule before.
+      cb(err);
```

Comment:
When passing explicit cb we have to emit `'error'` etc. in the callback. However, unless `cb` is invoked *first* it's not possible to schedule an `'error'` before `'close'`.